addons/event/models/event_ticket.py

```diff
@@ -111,10 +112,11 @@ def _compute_seats(self):
             self.env['event.registration'].flush(['event_id', 'event_ticket_id', 'state'])
             self.env.cr.execute(query, (tuple(self.ids),))
             for event_ticket_id, state, num in self.env.cr.fetchall():
-                ticket = self.browse(event_ticket_id)
-                ticket[state_field[state]] += num
+                results.setdefault(event_ticket_id, {})[state_field[state]] = num
```

Comment:
Shouldn't we do like event, prepare a dict with all fields at 0, and be sure to assign a value to all fields after computation ?

Reply:
Looking at line 98 it's not usefull I think.

Reply:
I think it is equivalent, since with 
https://github.com/odoo/odoo/blob/66d80865560fb6a0e3f00e2e566a5a3bb4138809/addons/event/models/event_ticket.py#L119
we do `ticket.update({})` if the ticket is not in results

Reply:
@nmarotte It's possible that for one ticket some fields are defined and other not in `results.get(ticket._origin.id or ticket.id, {})`

So `ticket.update(results.get(ticket._origin.id or ticket.id, {}))` won't update all fields.

but with line 97-98 ensure that all values are set to 0 at the beginning.

```
for ticket in self:
    ticket.seats_unconfirmed = ticket.seats_reserved = ticket.seats_used = ticket.seats_available = 0
```

Reply:
Oh indeed, we have a default value set beforehand.